tests/io_test.cpp

```diff
@@ -72,3 +74,97 @@ TEST_CASE("IoFreeWorkItem in worker", "[io]")
     Sleep(500);
     REQUIRE(context.value == 2);
 }
+
+struct _ex_pool_free_functor
+{
+    void
+    operator()(void* buffer)
+    {
+        ExFreePool(buffer);
+    }
+};
+
+struct _io_mdl_free_functor
+{
+    void
+    operator()(void* mdl)
+    {
+        IoFreeMdl((MDL*)mdl);
+    }
+};
+
+TEST_CASE("IoBuildPartialMdl", "[io]")
+{
+    const ULONG buffer_size = 8192;    // 2 pages
+    const ULONG partial_offset = 1024; // Offset into buffer
+    const ULONG partial_size = 2048;   // Partial size
+    ULONG tag = 'tset';
+
+    // Allocate buffer using ExAllocatePoolWithTag
```

Comment:
```suggestion
    // Allocate buffer using ExAllocatePoolWithTag.
```
Add periods at end of comments, for consistency with style in rest of repo.